lib/VeresOneClient.js

```diff
@@ -106,25 +106,32 @@ class VeresOneClient {
   }
 
   async getTicketServiceProof({operation, ticketService}) {
-    const baseURL = ticketService;
-    const ticketServiceApi = create({baseURL});
-    const response = await ticketServiceApi.post(
-      '/', {operation}, {httpsAgent: this.httpsAgent});
-    if(response.problem) {
+    let result;
+    try {
+      // note: ky will add / to the end of the url
+      result = await httpClient.post('', {
+        prefixUrl: ticketService,
+        json: {operation}, httpsAgent: this.httpsAgent
+      });
+    } catch(e) {
+      const {response} = e;
+      // if there is no response just rethrow the error
+      if(!response) {
+        throw e;
+      }
       const error = new VeresOneClientError(
         'Error retrieving record.', 'NetworkError');
-      if(response.problem === 'CLIENT_ERROR') {
-        error.details = {
-          baseURL, error: response.data, status: response.status
-        };
-      } else {
-        error.details = {
-          baseURL, error: response.originalError, status: response.status
-        };
+      error.details = {
+        baseURL: ticketService, error: e, status: response.status, result
+      };
+      // errors in this range might contain response data
+      if(response.status > 399 && response.status < 500) {
+        // httpClient puts the data directly on the error
+        error.details.error = e.data || response.data || e;
```

Comment:
Seems like the same helper that's needed in other places is needed here to DRY this up.

Reply:
issue here: https://github.com/veres-one/did-veres-one/issues/66